flusher.go

```diff
@@ -212,6 +214,33 @@ func (s *Server) reportGlobalMetricsFlushCounts(ms metricsSummary) {
 	s.statsd.Count("worker.metrics_flushed_total", int64(ms.totalTimers), []string{"metric_type:timer"}, 1.0)
 }
 
+func (s *Server) flushS3(finalMetrics []DDMetric) {
+	start := time.Now()
+
+	var data bytes.Buffer
+	err := json.NewEncoder(&data).Encode(finalMetrics)
+	if err != nil {
+		s.logger.WithFields(logrus.Fields{
+			"metrics": len(finalMetrics),
+		}).Error("Could not marshal finalMetrics before posting to s3")
+		return
+	}
+
+	// this feels dirty but oh well
+	seekableData := strings.NewReader(data.String())
```

Comment:
can we do `bytes.NewReader(data.Bytes())` instead? I think that way would avoid the slicebytetostring/stringtoslicebyte conversions